apache/superset
# fix(Explore): Show the tooltip only when label does not fit the container in METRICS/FILTERS/GROUP BY/SORT BY of the DATA panel

### SUMMARY
This PR makes the tooltip appear only when the label does not fit the container for the METRICS/FILTERS/GROUP BY (with D&D enabled)/SORT BY of the DATA panel. It also consolidates the look and behavior of the Columns and Metrics across the 3 different implementations, including the one in Superset UI. Finally, this PR hides the tooltip when the drag event starts.

###  REQUIRES SUPERSET-UI PR MERGED 
Attention, this PR requires the related PR https://github.com/apache-superset/superset-ui/pull/1271 in Superset-UI to be merged first.

### REFACTOR REQUIRED

Refactoring is suggested to remove the 3 different implementations and consolidate the code. This PR only intention is to resolve the issue #15996 within the limitations of the current implementation.

### BEFORE

https://user-images.githubusercontent.com/60598000/128191895-db410832-df25-4727-8df5-4579e0cb39a9.mp4

https://user-images.githubusercontent.com/60598000/128191922-1bae0671-33ab-4432-8c19-6ab6758cb03d.mp4

### AFTER

https://user-images.githubusercontent.com/60598000/128192168-606afa75-6634-48b9-b2e7-0e1036a2af5d.mp4

https://user-images.githubusercontent.com/60598000/128192223-080a903c-a103-4ae8-8a29-4273f4d8da88.mp4

### TESTING INSTRUCTIONS
1. Visit Explore
2. Add Columns and Metrics in Filters and Sort options either with drag and drop enabled or not
3. Observe the labels with long text and make sure they are truncated with ellipsis
4. Hover the labels and check the tooltip
5. Expand the panel until the text isn't truncated and hover the labels
6. Make sure the tooltip isn't appearing
7. Drag an item
8. Make sure the tooltip isn't appearing

### ADDITIONAL INFORMATION


- [x] Has associated issue: #15996
